pkg/manager/cert/cert_info.go

```diff
@@ -124,7 +126,12 @@ func (ci *certInfo) customizeTLSConfig(tlsConfig *tls.Config) *tls.Config {
 		}
 	} else {
 		tlsConfig.GetClientCertificate = func(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
-			return ci.getCertificate(), nil
+			cert := ci.getCertificate()
```

Comment:
Why not return the `emptyCert` in the `getCertificate`? It will also work when `ci.isServer==true`.

Reply:
`GetCertificate` and `GetClientCertificate` has different requirements. I mean `GetCertificate` should return nil.